clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -303,6 +318,20 @@ public PartitionConsumptionState(
     this.leaderCompleteState = LeaderCompleteState.LEADER_NOT_COMPLETED;
     this.lastLeaderCompleteStateUpdateInMs = 0;
     this.pendingReportIncPushVersionList = offsetRecord.getPendingReportIncPushVersionList();
+
+    KeyUrnCompressionDict keyUrnCompressionDict = offsetRecord.getKeyUrnCompressionDict();
+    if (keyUrnCompressionDict != null) {
+      if (keyUrnCompressionDict.keyUrnCompressionDictionaryVersion != 1) {
+        throw new VeniceException(
+            "Unsupported key urn compression dictionary version: "
+                + keyUrnCompressionDict.keyUrnCompressionDictionaryVersion);
+      }
+      UrnDictV1 urnDictV1 = UrnDictV1.loadDict(keyUrnCompressionDict.keyUrnCompressionDictionary);
+      this.keyUrnCompressor = new KeyUrnCompressor(
```

Comment:
I have a general question - why this Key compression is tied to a given partition? Could we do it in SIT level?

Reply:
There are two reasons:
1. The dict is not pre-built and we are building the dict along with the ingestion.
2. To avoid contentions among different partitions during ingestion.
Since the dict should be fairly small, keeping one dict per partition shouldn't introduce too much overhead.